synapse/handlers/profile.py

```diff
@@ -368,6 +372,39 @@ def set_avatar_url(self, target_user, requester, new_avatar_url, by_admin=False)
                 400, "Avatar URL is too long (max %i)" % (MAX_AVATAR_URL_LEN, ),
             )
 
+        # Enforce a max avatar size if one is defined
+        if self.max_avatar_size or self.allowed_avatar_mimetypes:
+            # Download the desired media file (possibly from a remote resource)
```

Comment:
Why would an avatar be a remote piece of media?

Reply:
Technically you can put any media in this field, remote or local. We *could* limit avatars to only local content, but you have to remember people can run media stores that are separate from Synapse as well.

Thus downloading the content seems to be the best way to handle this?

Reply:
Hmm, true. The problem is that we're downloading local content as if its remote, so we'll end up duplicating the media. I think its fair to require that user's avatars are local, and ideally if you have a separate media repo we'd somehow be able to query it. Given this is a) optional and b) for DINISC that don't use a separate media repo I'm tempted to punt that question down the road.

Reply:
Unfortunately we have expressed interest in porting this to mainline, but maybe that won't be possible given implementation constraints.

Reply:
TBH I think the lesser evil here is to say that if you want to use this option then you need to be using the synapse media repo, but possibly worth canvassing for opinions

Reply:
What would work better here in mainline land would be to land an endpoint where we can just ask the remote server for the image metadata, rather than downloading the image ourselves.

In this case, we can just go for the Synapse media repo only approach.

Reply:
I have done so.